datafusion/physical-expr/src/expressions/not.rs

```diff
@@ -104,14 +123,14 @@ pub fn not(
     arg: Arc<dyn PhysicalExpr>,
     input_schema: &Schema,
 ) -> Result<Arc<dyn PhysicalExpr>> {
-    let data_type = arg.data_type(input_schema)?;
-    if data_type != DataType::Boolean {
-        Err(DataFusionError::Internal(format!(
-            "NOT '{:?}' can't be evaluated because the expression's type is {:?}, not boolean",
+    match arg.data_type(input_schema)? {
+        // NULL is present as literal `ScalarValue::Utf8(None)` in logical expresion,
+        // to support `NOT(NULL)`, Utf8 arg is passed to `NotExpr` for further evaluate.
+        DataType::Boolean | DataType::Utf8 => Ok(Arc::new(NotExpr::new(arg))),
```

Comment:
What about detecting if the input is a `ScalarValue` and if that `ScalarValue::is_null()` returns true (rather than checking for utf8`?

Reply:
@alamb OK, will fix it.